exchanges/kucoin/kucoin_websocket.go

```diff
@@ -965,10 +965,12 @@ func (ku *Kucoin) processMarketSnapshot(respData []byte, instrument string) erro
 		LastUpdated:  response.Data.Datetime.Time(),
 	}
 	assetEnabledPairs := ku.listOfAssetsCurrencyPairEnabledFor(pair)
-	if assetEnabledPairs[asset.Spot] && ku.AssetWebsocketSupport.IsAssetWebsocketSupported(asset.Spot) && ku.CurrencyPairs.IsAssetEnabled(asset.Spot) == nil {
+	assetEnabledPairsSpot := assetEnabledPairs[asset.Spot] && ku.AssetWebsocketSupport.IsAssetWebsocketSupported(asset.Spot) && ku.CurrencyPairs.IsAssetEnabled(asset.Spot) == nil
```

Comment:
I think with the snapshot support now, we should be returning faster.
If you move the check before we make the ticker, we save allocations:
```go
	assetEnabledPairs := ku.listOfAssetsCurrencyPairEnabledFor(pair)
	assetEnabledPairsSpot := assetEnabledPairs[asset.Spot] && ku.AssetWebsocketSupport.IsAssetWebsocketSupported(asset.Spot) && ku.CurrencyPairs.IsAssetEnabled(asset.Spot) == nil
	assetEnabledPairsMargin := assetEnabledPairs[asset.Margin] && ku.AssetWebsocketSupport.IsAssetWebsocketSupported(asset.Margin) && ku.CurrencyPairs.IsAssetEnabled(asset.Margin) == nil
	if !assetEnabledPairsSpot && !assetEnabledPairsMargin {
		return nil
	}
     spotTickerPrice := ticker.Price{
...
...
...
```

Reply:
done